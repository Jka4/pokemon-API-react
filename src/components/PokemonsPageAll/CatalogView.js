import React from "react";
import 'react-tippy/dist/tippy.css'
import { setDelailedPageData } from "@APIutils";
import { NavLink } from "react-router-dom";

import './styles/style.scss';


const CatalogView = (props) => {
  const { pokemonDataArray } = props

  return (
    <div className="wrapper">
      {pokemonDataArray.map((index, key) => (
        <React.Fragment key={key}>
          <NavLink to={`/detailedPage/pokemon/${index?.name}`} data-pokemon_id={index?.id} onClick={setDelailedPageData}
            key={key}>
            <img src={index?.image} alt="" tabIndex={key} />
          </NavLink>
        </React.Fragment>
      ))
      }
    </div >
  )
}

export default React.memo(CatalogView);
