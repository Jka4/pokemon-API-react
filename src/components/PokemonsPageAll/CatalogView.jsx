import React, { Suspense, lazy } from "react";
import { NavLink } from "react-router-dom";
import PropTypes from "prop-types";

import { setDelailedPageData } from "@APIutils";

import "react-tippy/dist/tippy.css";
import "./styles/style.scss";

const ImageContainer = lazy(() => import("@ImageContainer"));

const CatalogView = ({ pokemonDataArray = [] }) => {
  return (
    <div className="wrapper">
      {pokemonDataArray.map((index, key) => (
        <React.Fragment key={key}>
          <NavLink
            to={`/detailedPage/pokemon/${index?.name}`}
            data-pokemon_id={index?.id}
            onClick={setDelailedPageData}
            key={key}
          >
            <Suspense
              fallback={
                <img
                  loading="lazy"
                  src={index?.placeholderBase64 || index?.image}
                  className="placeholderBase64"
                  alt="placeholderBase64"
                />
              }
            >
              <ImageContainer url={index?.image} cn="pokemonImageCard" />
            </Suspense>
          </NavLink>
        </React.Fragment>
      ))}
    </div>
  );
};

export default CatalogView;

CatalogView.propTypes = {
  pokemonDataArray: PropTypes.array,
};
