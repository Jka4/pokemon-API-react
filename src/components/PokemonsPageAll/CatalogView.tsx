import React, { lazy } from "react";
import { NavLink } from "react-router-dom";

import "react-tippy/dist/tippy.css";
import "./styles/style.scss";

const ImageContainer = lazy(() => import("components/ImageContainer/ImageContainer"));

type Props = {
  pokemonDataArray: {
    chain: {
      species_name: string;
    }[];
    id: 0;
    image: string;
    imageHQ: string;
    name: string;
    placeholderBase64: string;
    weight: number;
  }[];
}

const CatalogView: React.FC<Props> = ({ pokemonDataArray = [] }: Props) => {
  const fallback = (placeholderBase64: string) => {
    return (
      <>
        <img loading="lazy" src={placeholderBase64} className="placeholderBase64" alt="placeholderBase64" />
      </>
    );
  };

  return (
    <div className="wrapper">
      {pokemonDataArray.map((index, key) => (
        <React.Fragment key={key}>
          <NavLink to={`/detailedPage/pokemon/${index?.name}`} data-pokemon_id={index?.id} key={key} >
            <ImageContainer url={index?.image} cn="pokemonImageCard" fallback={fallback(index?.placeholderBase64)} />
          </NavLink>
        </React.Fragment>
      ))}
    </div>
  );
};

export default CatalogView;



