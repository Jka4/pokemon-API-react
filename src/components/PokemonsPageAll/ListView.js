import React, { lazy, Suspense, useState, useEffect } from "react";
import InfiniteScroll from "react-infinite-scroll-component";

import { setDelailedPageData } from "@APIutils";
import { NavLink } from "react-router-dom";
import Paper from "@material-ui/core/Paper";

import "./styles/style.scss";

const ImageContainer = lazy(() => import("@ImageContainer"));

const ListView = (props) => {
  const { pokemonDataArray } = props;

  const [pokemons, setPokemons] = useState([]);
  const [pokemonCount, setPokemonCount] = useState(0);

  useEffect(() => {
    fetchPokemons();

    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, []);

  const fetchPokemons = () => {
    const howMuchToDownload = 70;
    let arr = [];

    for (let i = pokemonCount; i <= pokemonCount + howMuchToDownload; i++) {
      arr.push(pokemonDataArray[i]);
    }
    setPokemons((pokemons) => [...pokemons, ...arr]);
    setPokemonCount(pokemons.length + howMuchToDownload);
  };

  return (
    <>
      <section className="main">
        <InfiniteScroll
          pageStart={1}
          dataLength={pokemons.length}
          next={fetchPokemons}
          hasMore={pokemons.length < 758}
          loader={<h4>Loading...</h4>}
          className="allPokemonsWrapper"
          endMessage={<p className="pokemon-end">No more pokemons :(</p>}
        >
          {pokemons.map((index, key) => (
            <NavLink
              to={`/detailedPage/pokemon/${index?.name}`}
              data-pokemon_id={index?.id}
              onClick={setDelailedPageData}
              key={index + Math.floor(1 + Math.random() * 9999999999)}
            >
              <Paper elevation={3} className="smallPokemonCard">
                <div className="pokemonLogo">
                  <Suspense
                    fallback={
                      <img
                        src={index?.placeholderBase64 || index?.image}
                        className="placeholderBase64"
                        alt="placeholderBase64"
                      />
                    }
                  >
                    <ImageContainer
                      url={index?.imageHQ || index?.image}
                      cn="pokemonImageCard"
                    />
                  </Suspense>
                </div>
                <div className="pokemonName">{index?.name}</div>
              </Paper>
            </NavLink>
          ))}
        </InfiniteScroll>
      </section>
    </>
  );
};

export default React.memo(ListView);
