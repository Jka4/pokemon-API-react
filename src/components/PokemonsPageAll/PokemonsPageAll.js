import React, { useEffect, lazy, Suspense, useState } from "react";
import Loader from 'react-loader-spinner'
import axios from 'axios';
import HeaderLine from "@HeaderLine";
import { Link } from "react-router-dom";

import './styles/style.scss';

const ImageContainer = lazy(() => import('@ImageContainer'));

const AllPokemonsPage = () => {
  const [pokemons, setPokemons] = useState([]);
  const [pokemonCount, setPokemonCount] = useState(0);

  useEffect(() => {
    fetchAllPokemons();
  }, []);

  const fetchAllPokemons = async () => {
    const howMuchToDownload = 40;

    for (let i = pokemonCount + 1; i <= pokemonCount + howMuchToDownload; i++) {
      const URL = `https://pokeapi.co/api/v2/pokemon/${i}/`;
      await axios.get(URL)
        .then(async (response) => {
          setPokemons(pokemons => [...pokemons, response.data]);
        })
        .catch((error) => {
          console.log(error);
        });
    }
    setPokemonCount(pokemons.length + howMuchToDownload)
  }

  const loadNext = () => {
    fetchAllPokemons();
  }

  return (
    <React.Fragment>
      <HeaderLine />
      <Link to='/' className='backToMainPage' > Back </Link>

      <section className='main'>
        <div className="allPokemonsWrapper"  >
          {pokemons.map((index, key) => (
            <div className="smallPokemonCard" key={index + Math.floor(1 + Math.random() * 9999999999)} >
              <div className="pokemonLogo">
                <Suspense fallback={<Loader type="TailSpin" height={90}
                  width={110} color={"red"}
                />}>
                  <ImageContainer url={index?.sprites?.front_default} />
                </Suspense>
              </div>
              <div className="pokemonName">{index?.name} {index?.id}</div>
            </div>
          ))}
        </div>
        <button className="loadNext" onClick={loadNext} >Load Next</button>
      </section>
    </React.Fragment>
  )
}

export default AllPokemonsPage;