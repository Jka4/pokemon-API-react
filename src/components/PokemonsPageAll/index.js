import React, { useState } from "react";
import { Provider, connect } from 'react-redux';

import store from "@Store";
import ListView from '@ListView';
import CatalogView from '@CatalogView';

const PokemonsPageAll = (props) => {
	const { pokemonDataArray } = props
	const [viewTypeIsCatalog, setViewTypeIsCatalog] = useState(true);

	return (
		<React.Fragment>
			<div className="viewType">
				<div className="viewTypeInner">
					<div className="innerBlock" onClick={() => setViewTypeIsCatalog(!viewTypeIsCatalog)} >
						<span>View type </span>
						<span className='separator' >|</span>

						{!viewTypeIsCatalog
							? (<div className="catalog">
								<svg viewBox="0 0 16 16" id="icon-view-less"><path fillRule="evenodd" clipRule="evenodd" d="M0 2C0 0.895416 0.895508 0 2 0C3.10449 0 4 0.895416 4 2C4 3.10458 3.10449 4 2 4C0.895508 4 0 3.10458 0 2ZM6 2C6 0.895416 6.89551 0 8 0C9.10449 0 10 0.895416 10 2C10 3.10458 9.10449 4 8 4C6.89551 4 6 3.10458 6 2ZM14 0C12.8955 0 12 0.895416 12 2C12 3.10458 12.8955 4 14 4C15.1045 4 16 3.10458 16 2C16 0.895416 15.1045 0 14 0ZM0 8C0 6.89542 0.895508 6 2 6C3.10449 6 4 6.89542 4 8C4 9.10458 3.10449 10 2 10C0.895508 10 0 9.10458 0 8ZM8 6C6.89551 6 6 6.89542 6 8C6 9.10458 6.89551 10 8 10C9.10449 10 10 9.10458 10 8C10 6.89542 9.10449 6 8 6ZM12 8C12 6.89542 12.8955 6 14 6C15.1045 6 16 6.89542 16 8C16 9.10458 15.1045 10 14 10C12.8955 10 12 9.10458 12 8ZM2 12C0.895508 12 0 12.8954 0 14C0 15.1046 0.895508 16 2 16C3.10449 16 4 15.1046 4 14C4 12.8954 3.10449 12 2 12ZM6 14C6 12.8954 6.89551 12 8 12C9.10449 12 10 12.8954 10 14C10 15.1046 9.10449 16 8 16C6.89551 16 6 15.1046 6 14ZM14 12C12.8955 12 12 12.8954 12 14C12 15.1046 12.8955 16 14 16C15.1045 16 16 15.1046 16 14C16 12.8954 15.1045 12 14 12Z"></path></svg>
							</div>)
							: (<div className="list">
								<svg viewBox="0 0 15 15" id="icon-view-more"><g><circle cx="1.5" cy="1.5" r="1.5"></circle><circle cx="1.5" cy="5.5" r="1.5"></circle><circle cx="1.5" cy="9.5" r="1.5"></circle><circle cx="5.5" cy="9.5" r="1.5"></circle><circle cx="9.5" cy="9.5" r="1.5"></circle><circle cx="13.5" cy="9.5" r="1.5"></circle><circle cx="13.5" cy="13.5" r="1.5"></circle><circle cx="9.5" cy="13.5" r="1.5"></circle><circle cx="5.5" cy="13.5" r="1.5"></circle><circle cx="1.5" cy="13.5" r="1.5"></circle><circle cx="5.5" cy="1.5" r="1.5"></circle><circle cx="5.5" cy="5.5" r="1.5"></circle><circle cx="9.5" cy="5.5" r="1.5"></circle><circle cx="13.5" cy="5.5" r="1.5"></circle><circle cx="9.5" cy="1.5" r="1.5"></circle><circle cx="13.5" cy="1.5" r="1.5"></circle></g></svg>
							</div>)}
					</div>
				</div>
			</div>

			{
				!viewTypeIsCatalog
					? (<ListView pokemonDataArray={pokemonDataArray} />)
					: (<CatalogView pokemonDataArray={pokemonDataArray} />)
			}

		</React.Fragment>
	)
}


const ConnectedPokemonsPageAll = connect(store => {
	return {
		pokemonDataArray: store.pokemonsArr,
	};
})(PokemonsPageAll);

export default props => (
	<Provider store={store}>
		<ConnectedPokemonsPageAll {...props} />
	</Provider>
);
