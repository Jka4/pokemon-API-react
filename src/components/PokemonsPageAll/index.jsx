/* eslint-disable import/no-anonymous-default-export */
import React, { useState } from "react";
import { Provider, connect } from "react-redux";
import PropTypes from "prop-types";

import store from "@Store";

import ListView from "@ListView";
import CatalogView from "@CatalogView";

import AppsIcon from "@material-ui/icons/Apps";
import BlurOnOutlinedIcon from "@material-ui/icons/BlurOnOutlined";

const PokemonsPageAll = ({ pokemonDataArray = [] }) => {
  const [viewTypeIsCatalog, setViewTypeIsCatalog] = useState(false);

  return (
    <>
      <div className="viewType">
        <div className="viewTypeInner">
          <div
            className="innerBlock"
            onClick={() => setViewTypeIsCatalog(!viewTypeIsCatalog)}
          >
            {!viewTypeIsCatalog ? (
              <>
                <span>List view:</span>
                <span className="separator">|</span>
                <div className="list">
                  <BlurOnOutlinedIcon />
                </div>
              </>
            ) : (
              <>
                <span>Catalog view:</span>
                <span className="separator">|</span>

                <div className="catalog">
                  <AppsIcon />
                </div>
              </>
            )}
          </div>
        </div>
      </div>

      {!viewTypeIsCatalog ? (
        <ListView pokemonDataArray={pokemonDataArray} />
      ) : (
        <CatalogView pokemonDataArray={pokemonDataArray} />
      )}
    </>
  );
};

const ConnectedPokemonsPageAll = connect((store) => {
  return {
    pokemonDataArray: store.pokemonsArr,
  };
})(PokemonsPageAll);

export default (props) => (
  <Provider store={store}>
    <ConnectedPokemonsPageAll {...props} />
  </Provider>
);

PokemonsPageAll.propTypes = {
  pokemonDataArray: PropTypes.array,
};
