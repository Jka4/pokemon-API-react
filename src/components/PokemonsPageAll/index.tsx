/* eslint-disable import/no-anonymous-default-export */
import React, { useState } from "react";
import { Provider, connect } from "react-redux";

import store from "../../Store/store";

import ListView from "./ListView";
import CatalogView from "./CatalogView";

import AppsIcon from "@material-ui/icons/Apps";
import BlurOnOutlinedIcon from "@material-ui/icons/BlurOnOutlined";

type PokemonsPageAllType = {
  pokemonDataArray: {
    chain: {
      species_name: string;
    }[];
    id: 0;
    image: string;
    imageHQ: string;
    name: string;
    placeholderBase64: string;
    weight: number;
  }[];
};

const PokemonsPageAll: React.FC<PokemonsPageAllType> = ({ pokemonDataArray = [] }: PokemonsPageAllType) => {
  const [viewTypeIsCatalog, setViewTypeIsCatalog] = useState<boolean>(false);

  return (
    <>
      <div className="viewType">
        <div className="viewTypeInner">
          <div
            className="innerBlock"
            onClick={() => setViewTypeIsCatalog(!viewTypeIsCatalog)}
          >
            {!viewTypeIsCatalog ? (
              <>
                <span>List view:</span>
                <span className="separator">|</span>
                <div className="list">
                  <BlurOnOutlinedIcon />
                </div>
              </>
            ) : (
                <>
                  <span>Catalog view:</span>
                  <span className="separator">|</span>

                  <div className="catalog">
                    <AppsIcon />
                  </div>
                </>
              )}
          </div>
        </div>
      </div>

      {!viewTypeIsCatalog ? (
        <ListView pokemonDataArray={pokemonDataArray} />
      ) : (
          <CatalogView pokemonDataArray={pokemonDataArray} />
        )}
    </>
  );
};

const ConnectedPokemonsPageAll = connect((store: { pokemonsArr: any[] }) => {
  return {
    pokemonDataArray: store.pokemonsArr,
  };
})(PokemonsPageAll);

export default (props = {}) => (
  <Provider store={store}>
    <ConnectedPokemonsPageAll {...props} />
  </Provider>
);
