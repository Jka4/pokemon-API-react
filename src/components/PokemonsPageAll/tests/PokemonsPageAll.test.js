import React from "react";
import { render } from "@testing-library/react";
import PokemonsPageAll from "@PokemonsPageAll";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";

it("AllPokemonsPage renders correctly", async () => {
  const tree = renderer
    .create(
      <Router>
        <PokemonsPageAll />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
