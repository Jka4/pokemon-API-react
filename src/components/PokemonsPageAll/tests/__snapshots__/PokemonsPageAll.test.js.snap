// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PokemonsPageAll AllPokemonsPage renders correctly 1`] = `
Array [
  <div
    className="viewType"
  >
    <div
      className="viewTypeInner"
    >
      <div
        className="innerBlock"
        onClick={[Function]}
      >
        <span>
          View type 
        </span>
        <span
          className="separator"
        >
          |
        </span>
        <div
          className="catalog"
        >
          <svg
            id="icon-view-less"
            viewBox="0 0 16 16"
          >
            <path
              clipRule="evenodd"
              d="M0 2C0 0.895416 0.895508 0 2 0C3.10449 0 4 0.895416 4 2C4 3.10458 3.10449 4 2 4C0.895508 4 0 3.10458 0 2ZM6 2C6 0.895416 6.89551 0 8 0C9.10449 0 10 0.895416 10 2C10 3.10458 9.10449 4 8 4C6.89551 4 6 3.10458 6 2ZM14 0C12.8955 0 12 0.895416 12 2C12 3.10458 12.8955 4 14 4C15.1045 4 16 3.10458 16 2C16 0.895416 15.1045 0 14 0ZM0 8C0 6.89542 0.895508 6 2 6C3.10449 6 4 6.89542 4 8C4 9.10458 3.10449 10 2 10C0.895508 10 0 9.10458 0 8ZM8 6C6.89551 6 6 6.89542 6 8C6 9.10458 6.89551 10 8 10C9.10449 10 10 9.10458 10 8C10 6.89542 9.10449 6 8 6ZM12 8C12 6.89542 12.8955 6 14 6C15.1045 6 16 6.89542 16 8C16 9.10458 15.1045 10 14 10C12.8955 10 12 9.10458 12 8ZM2 12C0.895508 12 0 12.8954 0 14C0 15.1046 0.895508 16 2 16C3.10449 16 4 15.1046 4 14C4 12.8954 3.10449 12 2 12ZM6 14C6 12.8954 6.89551 12 8 12C9.10449 12 10 12.8954 10 14C10 15.1046 9.10449 16 8 16C6.89551 16 6 15.1046 6 14ZM14 12C12.8955 12 12 12.8954 12 14C12 15.1046 12.8955 16 14 16C15.1045 16 16 15.1046 16 14C16 12.8954 15.1045 12 14 12Z"
              fillRule="evenodd"
            />
          </svg>
        </div>
      </div>
    </div>
  </div>,
  <section
    className="main"
  >
    <div
      className="infinite-scroll-component__outerdiv"
      style={Object {}}
    >
      <div
        className="infinite-scroll-component allPokemonsWrapper"
        style={
          Object {
            "WebkitOverflowScrolling": "touch",
            "height": "auto",
            "overflow": "auto",
          }
        }
      >
        <h4>
          Loading...
        </h4>
      </div>
    </div>
  </section>,
]
`;
